src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -339,10 +362,22 @@ private String locateNoCheck(String name) {
      */
     @Nullable
     private String locate(String name) {
-      String path = locateNoCheck(name);
-      if (modulePaths.contains(path)) {
-        return path;
+      InputPathData pathData = new InputPathData(canonicalizePath(name));
+      InputPathData modulePath = modulePaths.get(pathData.ensureLeadingSlash());
+      if (modulePath != null) {
```

Comment:
Please add comments to explain.
e.g. If we find the module by forcing the path to be absolute, why do we return the path unmodified?